scripts/new_project.kts

```diff
@@ -10,37 +10,75 @@ object NewProject {
     private const val TEMPLATE_FOLDER_NAME = "CoroutineTemplate"
     private const val TEMPLATE_PACKAGE_NAME = "co.nimblehq.coroutine"
 
+    private const val PATTERN_APP = "^[A-Z\$][a-zA-Z0-9(\\s)\$]*\$"
```

Comment:
@Tuubz If we don't have `\` if will be just `string` in Kotlin?

Reply:
> If we don't need it, I think the following is sufficient: ^[a-zA-Z0-9\\s]*$

The first character should be Uppercase 🤔

Reply:
Actually, something like `toyota-android-wallet` should also be allowed 💭 I just tried with the old script, but then the application class name will be renamed to `toyota-android-walletApplication` 🙈 

I think let's require uppercase for now, later we can make adjustments to allow cases like `toyota-android-wallet`.

What do you think? 🙏

Reply:
> @Tuubz If we don't have \ it will be just string in Kotlin?

Sorry, I don't think I explained myself well enough

From my understanding `\$` means that `$` is allowed as character, but that's not the case, right? 🙏 

So I think the end result should be `"^[A-Z][a-zA-Z0-9\\s]*$"`, please correct me if I'm wrong 🙇

Reply:
> From my understanding \$ means that $ is allowed as character, but that's not the case, right? 🙏

@Tuubz You mean (`s` or `S`) not `$` right? 
![Screen Shot 2565-05-25 at 13 42 10](https://user-images.githubusercontent.com/28002315/170197884-9031b238-211d-4966-bab7-e130a1587b06.png)

Reply:
I see that you also mentioned `\\s`, this part is correct 👍 

But I'm referring to these:

![Screen Shot 2565-05-25 at 14 44 10](https://user-images.githubusercontent.com/18277915/170208988-0093a29f-8513-446b-bdcd-5f25c1864aee.png)

From my understanding, they should be removed. Because this shouldn't be allowed `app-name="Test($)"` 🙏

Reply:
>Actually, something like toyota-android-wallet should also be allowed 💭 I just tried with the old script, but then the application class name will be renamed to toyota-android-walletApplication 🙈 I think let's require uppercase for now, later we can make adjustments to allow cases like toyota-android-wallet.
What do you think? 🙏

Yeah, I agreed, Let require uppercase for now and you can create the issue to solved this later if need  🚀